### PR TITLE
[A2-572] Add 'viewer' and 'editor' users to local-dev and dev/acceptance envs, add `--skip-policy-migration` to `upgrade-to-v2`

### PR DESCRIPTION
**Pull request by srenatus**
### :nut_and_bolt: Description

Our environments, both for local development and dev/acceptance, are less useful than they could be if everyone keeps using the admin user for all the things. Especially in the short-lived ones, it's cumbersome to add these users manually for acceptance testing time and time again.

The same goes for the local development environment. The world looks quite different if you're not an admin 😉

~All of this only applies to envs using IAM v2 or v2.1. However, the users can exist in v1 environments in just the same way, so the local studio env is *not* upgraded to IAM v2 or IAM v2.1 _yet_ (as of now).~ ✔️ local studio env is upgrade as well.

#### Reviewers!

Please follow the commits. There's a few of them, and they should tell the story of this PR 😉 

### :+1: Definition of Done

- [X] In all used environments (IAM v2 or IAM v2.1), there are these two default users, members of their corresponding teams (`team:local:viewers`, `team:local:editors`)
- [X] the `a2-iam-v2-only-integration` inspec suite no longer creates the users itself, but relies on the newly added cli command for that
- [X] there's a (hidden) flag to `chef-automate iam upgrade-to-v2`, `--skip-legacy-upgrade`, that causes v1 policies to _not_ get migrated
- [x] that flag is used for our short-lived dev/acceptance envs
- [x] that flag is used in `start_all_services`
- [x] rename the flag to `--skip-policy-migration` for clarity (discussed this with @susanev)

#### local development

- `rebuild components/automate-cli`
- `rebuild components/automate-gateway`, `rebuild components/authz-service` for testing `chef-automate iam update-to-v2 --skip-policy-migration`
- `start_all_services`
- login as either "viewer" or "editor" (password is "chefautomate")

- to independently check the `--skip-policy-migration` flag's output:
```
[78][default:/src:0]# chef-automate iam reset-to-v1

Reverting to IAM v1 (with pre-upgrade v1 policies)
[79][default:/src:0]# chef-automate iam upgrade-to-v2 --beta2.1 --skip-policy-migration

Enabling IAM v2.1

Skipped policies:
  28 v1 policies
Creating default teams Editors and Viewers...
Skipped: Editors team already exists
Skipped: Viewers team already exists

Migrating existing teams...

Success: Enabled IAM v2.1
[80][default:/src:0]#
```
- note that for now, the skip flag is **unhidden** in the dev env (studio):
```
[8][default:/src:0]# chef-automate iam upgrade-to-v2 --help
Upgrade to IAM v2 and migrate existing v1 policies. On downgrade, any new v2 policies will be reverted.

Usage:
  chef-automate iam upgrade-to-v2 [flags]

Flags:
      --beta2.1                 Upgrade to version 2.1 with beta project authorization.
  -h, --help                    help for upgrade-to-v2
      --skip-policy-migration   Do not migrate policies from IAM v1.

Global Flags:
  -d, --debug                Enable debug output
      --no-check-version     Disable version check
      --result-json string   Write command result as JSON to PATH
```
- to simulate outside-of-dev behaviour, set `CHEF_DEV_ENVIRONMENT` to something that's not `yes`:
```
[80][default:/src:0]# env CHEF_DEV_ENVIRONMENT=no chef-automate iam upgrade-to-v2 --help
Upgrade to IAM v2 and migrate existing v1 policies. On downgrade, any new v2 policies will be reverted.

Usage:
  chef-automate iam upgrade-to-v2 [flags]

Flags:
  -h, --help   help for upgrade-to-v2

Global Flags:
  -d, --debug                Enable debug output
      --no-check-version     Disable version check
      --result-json string   Write command result as JSON to PATH
```

#### dev/acceptance envs

- let's check that post-merge and post-promotion 🤞 

### :chains: Related Resources

- [A2-572](https://chefio.atlassian.net/browse/A2-572)

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [X] Code actually executed?
- [X] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/iam.go [hunk 1]
```diff
@@ -93,17 +95,20 @@ func newIAMUpgradeToV2Cmd() *cobra.Command {
...
 		"Upgrade to version 2.1 with beta project authorization.")
-	err := cmd.PersistentFlags().MarkHidden("beta2.1")
-	// we could also ignore the lint error :shrug:
-	if err != nil {
-		fmt.Printf("failed configuring cobra: %s\n", err.Error())
-		panic(err.Error())
-	}
+
+	// all flags are hidden right now
+	cmd.PersistentFlags().VisitAll(func(f *pflag.Flag) { f.Hidden = !isDevMode() })
```
Comment (msorens) [inline components/automate-cli/cmd/chef-automate/iam.go]: I thought the team consensus was that `skip-policy-migration` was for public consumption as well...?

Comment (susanev) [reply components/automate-cli/cmd/chef-automate/iam.go]: i believe that consensus convo happened last friday during subtasking and i wasnt there. if we do make it public we should have a corresponding docs update, so id like to do that in a separate pr if possible.

Comment (srenatus) [reply components/automate-cli/cmd/chef-automate/iam.go]: I'm all in favour of not adding anything here 😉

File: terraform/test-environments/modules/chef_automate_install/templates/install_chef_automate_cli.sh.tpl [hunk 2]
```diff
@@ -130,11 +130,13 @@ hab pkg binlink core/hab --force
...
     "v2.1")
-      chef-automate iam upgrade-to-v2 --beta2.1
+      chef-automate iam upgrade-to-v2 --beta2.1 --skip-policy-migration
+      chef-automate dev create-iam-dev-users
       touch /root/a2-iamv2-enabled
       ;;
     "v2")
-      chef-automate iam upgrade-to-v2
+      chef-automate iam upgrade-to-v2 --skip-policy-migration
+      chef-automate dev create-iam-dev-users
```
Comment (msorens) [inline terraform/test-environments/modules/chef_automate_install/templates/install_chef_automate_cli.sh.tpl]: Good stuff!

File: components/automate-cli/cmd/chef-automate/dev.go [hunk 3]
```diff
@@ -974,3 +986,35 @@ func runDisablePrometheusCmd(*cobra.Command, []string) error {
...
+		return err
+	}
+	for username, data := range map[string]struct {
+		displayName, password, team string
+	}{
+		"viewer": {"Viewer User", "chefautomate", "viewers"},
+		"editor": {"Editor User", "chefautomate", "editors"},
+	} {
+		userID, _, err := adminmgmt.CreateUserOrUpdatePassword(ctx,
+			apiClient, username, data.displayName, data.password, false /* dry run */)
```
[the remark this answers is not in the document]
Comment (srenatus) [reply components/automate-cli/cmd/chef-automate/dev.go]: hehe I didn't mean to convey that much -- I really just meant "this `bool` is the dry run parameter". But yeah, you're not wrong with that connotation 😉

File: components/authz-service/server/v2/policy.go [hunk 4]
```diff
@@ -556,14 +557,27 @@ func (s *policyServer) MigrateToV2(ctx context.Context,
...
+	} else {
+		// Note 2019/05/22 (sr): policies without subjects are silently ignored -- this
+		// is to be in line with the migration case, that does the same. However, this
+		// could be worth revisiting?
+		pols, err := s.v1.ListPoliciesWithSubjects(ctx)
+		if err != nil {
+			recordFailure()
+			return nil, status.Errorf(codes.Internal, "list v1 policies: %s", err.Error())
+		}
+		reports = append(reports, fmt.Sprintf("%d v1 policies", len(pols)))
```
[the remark this answers is not in the document]
Comment (srenatus) [reply components/authz-service/server/v2/policy.go]: The output, in the end, will be

```
[79][default:/src:0]# chef-automate iam upgrade-to-v2 --beta2.1 --skip-policy-migration

Enabling IAM v2.1

Skipped policies:
  28 v1 policies
```
(see PR message) -- I'd rather not use skipped twice there

[the remark this answers is not in the document]
Comment (susanev) [reply components/authz-service/server/v2/policy.go]: @srenatus do you mind if we just have a single line with `Skipped: %d v1 policies`, then a newline before the teams output?

[the remark this answers is not in the document]
Comment (srenatus) [reply components/authz-service/server/v2/policy.go]: Right now, that's shared between a migration and no migration (see https://github.com/chef/a2/pull/4120#issuecomment-433389839), so it's a little more work than just the formatting. But yeah, we can do that.

[the remark this answers is not in the document]
Comment (susanev) [reply components/authz-service/server/v2/policy.go]: oh i see.... nah, just leave it as is now.

**Comment by srenatus** [comment]
I'll merge this on Monday and deal with the fallout.

**Pull request merged**